src/crafting.cpp

```diff
@@ -1746,85 +1785,76 @@ void set_item_inventory(item &newit)
         bool pl = false, mp = false;
 
         if (item::count_by_charges(type) && count > 0) {
             if (has_charges(type, count)) {
-                player_has.push_back( component );
+                player_has.push_back(component);
                 pl = true;
             }
             if (map_inv.has_charges(type, count)) {
-                map_has.push_back( component );
+                map_has.push_back(component);
                 mp = true;
             }
             if (!pl && !mp && charges_of(type) + map_inv.charges_of(type) >= count) {
-                mixed.push_back( component );
+                mixed.push_back(component);
             }
-        } else { // Counting by units, not charges
+        }else { // Counting by units, not charges
 
             if (has_amount(type, count)) {
-                player_has.push_back( component );
+                player_has.push_back(component);
                 pl = true;
             }
             if (map_inv.has_components(type, count)) {
-                map_has.push_back( component );
+                map_has.push_back(component);
                 mp = true;
             }
             if (!pl && !mp && amount_of(type) + map_inv.amount_of(type) >= count) {
-                mixed.push_back( component );
+                mixed.push_back(component);
             }
 
         }
     }
 
+    /* select 1 component to use */
     if (player_has.size() + map_has.size() + mixed.size() == 1) { // Only 1 choice
         if (player_has.size() == 1) {
-            use_from = use_from_player;
-            selected_comp = player_has[0];
-        } else if (map_has.size() == 1) {
-            use_from = use_from_map;
-            selected_comp = map_has[0];
+            selected.use_from = usage::use_from_player;
+            selected.comp = player_has[0];
+        }
+        else if (map_has.size() == 1) {
+            selected.use_from = usage::use_from_map;
+            selected.comp = map_has[0];
         } else {
-            use_from = use_from_both;
-            selected_comp = mixed[0];
+            selected.use_from = usage::use_from_both;
+            selected.comp = mixed[0];
         }
     } else { // Let the player pick which component they want to use
         std::vector<std::string> options; // List for the menu_vec below
-        // Populate options with the names of the items
-        for( auto &map_ha : map_has ) {
-            std::string tmpStr = item::nname( map_ha.type ) + _( " (nearby)" );
+                                          // Populate options with the names of the items
+        for (auto &map_ha : map_has) {
+            std::string tmpStr = item::nname(map_ha.type) + _(" (nearby)");
             options.push_back(tmpStr);
         }
-        for( auto &player_ha : player_has ) {
-            options.push_back( item::nname( player_ha.type ) );
+        for (auto &player_ha : player_has) {
```

Comment:
It was established after most of the code had already been written. The existing code wasn't touched at that point because it would make too many problems. The current rule is to formatted new code and format code that is change anyway.